### PR TITLE
Wait for the host agent process to start in host agent integration tests

### LINKED ISSUE #438: Flaky Host Agent Integration tests
**What steps did you take and what happened:**
The host agent integration tests are quite flaky and are intermittently failing in the CI as well as local environments.
Here is one such failing CI example -  https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/runs/5570549277?check_suite_focus=true

The tests in this file need to revisited - https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/agent/host_agent_test.go

### PR DESCRIPTION
**Pull request by anusha94**
Since we are running the host agent in the containers for integration
tests, the process is not started instantly sometimes and this causes
flakes in the tests. Adding a check to ensure the host agent process is
started and only then proceed with the tests

Signed-off-by: Nilanjan Daw <dawn@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #438

### CODE REVIEW

File: agent/host_agent_test.go [hunk 1]
```diff
@@ -157,6 +157,18 @@ var _ = Describe("Agent", func() {
...
+			Eventually(func() bool {
+				containerTop, _ := runner.DockerClient.ContainerTop(ctx, byoHostContainer.ID, []string{})
+				for _, proc := range containerTop.Processes {
+					if strings.Contains(proc[len(containerTop.Titles)-1], "agent") {
+						return true
+					}
+
+				}
+				return false
+			}).Should(BeTrue())
```
Comment (dharmjit) [inline agent/host_agent_test.go:171]: Do we set an explicit timeout interval for this `eventually` block? This might fail as the default timeout is 1 sec.

Comment (anusha94) [reply agent/host_agent_test.go:171]: Good point. What do you think is a reasonable timeout for this? `60s`?

**Review by dharmjit — APPROVED** [review]
lgtm

**Comment by jamiemonserrate** [comment]
I think this makes sense, and is a good incremental step to improving the flakiness. I am _slightly_ worried that this check might pass wrongly if there is _any_ process that matches the name `agent`. But this is a good start. 

But is this enough? That is, should we be checking if the host-agent has reached a desired state? Like maybe the log messages should say "agent started/ready" or something?

Just a suggestion, as I'm not sure what exactly we need to wait for.

**Review by jamiemonserrate — APPROVED** [review]

**Comment by anusha94** [comment]
> But is this enough? That is, should we be checking if the host-agent has reached a desired state? Like maybe the log messages should say "agent started/ready" or something?

Good suggestion, if the flakiness persists, we could try that.

**Pull request merged**